File: ruby/item.rb
class Item
  attr_accessor :name, :sell_in, :quality

  def initialize(name, sell_in, quality)
    @name = name  # Name of the item
    @sell_in = sell_in  # Number of days to sell the item
    @quality = quality  # Represents how valuable the item is
  end

  def to_s()
    "#{@name}, #{@sell_in}, #{@quality}"  # Representing the item's details as a string
  end
end

File: TypeScript/app/item.ts
export class Item {
  name: string;
  sellIn: number;
  quality: number;

  // can't edit this constructor because of the kata rules.
  // but I would change the constructor to take a javaScript object.
  // It makes the consumer code more readable.
  // e.g. new Item({ name: "standard item", sellIn: 0, quality: 2 })
  // instead of new Item("standard item", 0, 2)

  // I would also type the parameters, because now they are implicitly any.
  constructor(name, sellIn, quality) {
    this.name = name;
    this.sellIn = sellIn;
    this.quality = quality;
  }
}
